dali/pipeline/operator/op_schema.h

```diff
@@ -493,384 +493,360 @@ used with DALIDataType, to avoid confusion with `AddOptionalArg<type>(name, doc,
   void InitNames();
 
   std::string dox_;
-  /** @brief The name of the schema */
+  /// The name of the schema
   std::string name_;
-  /** @brief The module path for the operator */
+  /// The module path for the operator
   std::vector<std::string> module_path_;
-  /** @brief The camel case name of the operator (without the module path) */
+  /// The PascalCase name of the operator (without the module path)
   std::string operator_name_;
 
-  bool disable_auto_input_dox_ = false;
+  ////////////////////////////////////////////////////////////////////////////
+  // Inputs, outputs and arguments
 
-  struct InputDoc {
-    std::string name = {};
-    std::string type_doc = {};
-    std::string doc = {};
-  };
-  std::vector<InputDoc> input_dox_ = {};
-  bool input_dox_set_ = false;
+  /// All locally defined arguments
+  std::map<std::string, ArgumentDef, std::less<>> arguments_;
 
-  // Custom docstring, if not empty should be used in place of input_dox_ descriptions
-  std::string call_dox_ = {};
+  mutable
+  detail::LazyValue<std::map<std::string, const ArgumentDef *, std::less<>>> flattened_arguments_;
+  mutable int circular_inheritance_detector_ = 0;
 
-  // Whether to append kwargs section to __call__ docstring. Off by default,
-  // can be turned on for call_dox_ specified manually
-  bool append_kwargs_section_ = false;
+  std::map<std::string, const ArgumentDef *, std::less<>> &GetFlattenedArguments() const;
+
+  /// The properties of the inputs
+  std::vector<InputInfo> input_info_;
+  bool disable_auto_input_dox_ = false;
+  bool input_dox_set_ = false;
 
   SpecFunc output_fn_, in_place_fn_, additional_outputs_fn_;
 
   int min_num_input_ = 0, max_num_input_ = 0;
   int num_output_ = 0;
 
-  bool allow_instance_grouping_ = true;
-  vector<string> parents_;
+  ////////////////////////////////////////////////////////////////////////////
+  // Schema inheritance
 
-  bool support_volumetric_ = false;
+  /// Names of the parent schemas
+  vector<string> parent_names_;
+  /// Cached pointers to parent schemas, to avoid repeated lookups
+  mutable detail::LazyValue<std::vector<const OpSchema *>> parents_;
 
+  ////////////////////////////////////////////////////////////////////////////
+  // Documentation-related
+  bool support_volumetric_ = false;
   bool allow_sequences_ = false;
   bool is_sequence_operator_ = false;
 
   bool is_internal_ = false;
   bool is_doc_hidden_ = false;
   bool is_doc_partially_hidden_ = false;
 
-  bool no_prune_ = false;
+  /// Custom docstring, if not empty should be used in place of input_dox_ descriptions
+  std::string call_dox_ = {};
+
+  /// Whether to append kwargs section to __call__ docstring. Off by default,
+  /// can be turned on for call_dox_ specified manually
+  bool append_kwargs_section_ = false;
 
+  ////////////////////////////////////////////////////////////////////////////
+  // Internal flags
+  bool allow_instance_grouping_ = true;
```

Comment:
nitpick - does anything every really happens to it, I did not find any DisallowInstanceGrouping uses.

Reply:
Removed.